action.yml

```diff
@@ -13,36 +21,19 @@ runs:
   using: "composite"
   steps:
     - name: Nix dead code analysis
-      run: |
-        TMP=$(mktemp -d)
-        nix build -o $TMP/deadnix github:astro/deadnix
-        MSG="$($TMP/deadnix/bin/deadnix ${{ inputs.flags }} -e .)"
-        MSG="${MSG//'%'/'%25'}"
-        MSG="${MSG//$'\n'/'%0A'}"
-        MSG="${MSG//$'\r'/'%0D'}"
-        echo -n "::set-output name=msg::$MSG"
+      run: 'nix run nixpkgs#deadnix -- ${{ inputs.flags }} ${{ inputs.directory }}'
       shell: bash
       id: deadnix
-    - run: |
-        if ! git diff --quiet --exit-code ; then
-          git commit -a -F - <<EOF
-          ${{ inputs.commit_message }}
-        EOF
-        fi
-      env:
-        GIT_AUTHOR_NAME: github-actions[bot]
-        GIT_AUTHOR_EMAIL: <github-actions[bot]@users.noreply.github.com>
-        GIT_COMMITTER_NAME: github-actions[bot]
-        GIT_COMMITTER_EMAIL: <github-actions[bot]@users.noreply.github.com>
-      shell: bash
     - name: Create PR
-      uses: peter-evans/create-pull-request@v3
+      uses: peter-evans/create-pull-request@v4
+      if: inputs.create_pr == 'true'
       with:
         branch: deadnix
         delete-branch: true
         title: "${{ inputs.commit_message }}"
+        commit-message: "${{ inputs.commit_message }}"
         body: |
           Automated changes by the [deadnix-action](https://github.com/astro/deadnix-action) GitHub Action.
 branding:
-    icon: 'minus-circle'
-    color: 'red'
+    icon: 'check-circle'
```

Comment:
Minor nitpick: why change the logo to a `'check-circle'`? I chose `'minus-circle'` because the tool doesn't run tests but tries to remove code.

Reply:
Done.